### PR TITLE
[E-idora < T391] Implement from pairs for maps

### PR DESCRIPTION
**Pull request by ind1xa**
### Description

Implemented map.fromPairs procedure and e2e tests for it

### Pull request type

- [ ] Bugfix
- [x ] Algorithm/Module
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

######################################

#### Module/Algorithm
- [ ] Core algorithm/module implementation
- [ ] [Query module](https://memgraph.com/docs/memgraph/reference-guide/query-modules) implementation
- [ ] Unit tests
- [ ] End-to-end tests
- [ ] Code documentation
- [ ] README short description
- [ ] Documentation on [memgraph/docs](https://github.com/memgraph/docs)
- [ ] Update GQLALchemy signatures in [query builder](https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/graph_algorithms/query_builder.py  ) using [query module signature generator](https://github.com/memgraph/gqlalchemy/blob/main/scripts/query_module_signature_generator.py)

######################################

### CODE REVIEW

File: cpp/map_module/algorithm/map.cpp [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#include "map.hpp"
+
+void Map::FromPairs(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const auto &list = arguments[0].ValueList();
+
+    mgp::Map pairsMap;
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:13]: `pairs_map`, we need to be consistent with naming

File: cpp/map_module/algorithm/map.cpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#include <sstream>
+
+#include "map.hpp"
+
+void Map::FromPairs(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::memory = memory;
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const auto &list = arguments[0].ValueList();
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:11]: Remove reference here, and just use `const auto`, as `ValueList` returns temporary, we discussed that today on Slack. I think we will update that on Epic branch for others then

File: cpp/map_module/algorithm/map.cpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    mgp::Map pairsMap;
+
+    for (const auto inside_list : list) {
+      if (inside_list.ValueList().Size() != 2) {
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:16]: 2 is magic number here, check following: https://stackoverflow.com/questions/47882/what-is-a-magic-number-and-why-is-it-bad

Comment (antoniofilipovic) [reply cpp/map_module/algorithm/map.cpp:16]: Isolate that number somewhere

File: cpp/map_module/algorithm/map.cpp [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        throw mgp::IndexException("Pairs must consist of 2 elements exactly.");
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:17]: You can use this messaage `fmt::format` with number 2

File: cpp/map_module/algorithm/map.cpp (hunk 2 above)
Comment (antoniofilipovic) [reply cpp/map_module/algorithm/map.cpp:11]: https://memgraph.slack.com/archives/C05G7B5V1TR/p1690361110221839

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few changes to do

**Review by antoniofilipovic — APPROVED** [review]
Seems good

File: cpp/map_module/algorithm/map.cpp [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const auto list = arguments[0].ValueList();
+
+    mgp::Map pairs_map;
+
+    for (const auto inside_list : list) {
+      if (inside_list.ValueList().Size() != number_of_elements_in_pair) {
+        throw mgp::IndexException("Number of elements in a pair is not right.");
```
Comment (antoniofilipovic) [inline cpp/map_module/algorithm/map.cpp:19]: Did you try to use the `fmt::format` library here?

Comment (antoniofilipovic) [reply cpp/map_module/algorithm/map.cpp:19]: It would be helpfult to output 2 as number

Comment (ind1xa) [reply cpp/map_module/algorithm/map.cpp:19]: sorry I wasn't sure how to work with it... when I install that library locally do I somehow have to include it in this PR?

Comment (antoniofilipovic) [reply cpp/map_module/algorithm/map.cpp:19]: # Add all module files related to graph util module
set(collections_module_src
    collections_module.cpp ...)

add_query_module(collections 1 "${collections_module_src}")

target_link_libraries(collections_module_src PRIVATE fmt::fmt) <- try with this

**Pull request merged**